plugins/modules/vmware_guest.py

```diff
@@ -3231,20 +3274,29 @@ def reconfigure_vm(self):
                 uuid_action_opt.value = "create"
                 self.configspec.extraConfig.append(uuid_action_opt)
 
+            # TODO latent bug? self.change_detected isn't checked again after this
+            # Does the configspec.extraConfig uuid_action_opt append above ever get used?
             self.change_detected = True
 
         # add customize existing VM after VM re-configure
         if self.params['customization']['existing_vm']:
             if self.current_vm_obj.config.template:
                 self.module.fail_json(msg="VM is template, not support guest OS customization.")
-            if self.current_vm_obj.runtime.powerState != vim.VirtualMachinePowerState.poweredOff:
+            if self.current_vm_obj.runtime.powerState != vim.VirtualMachinePowerState.poweredOff and not self.module.check_mode:
                 self.module.fail_json(msg="VM is not in poweroff state, can not do guest OS customization.")
-            cus_result = self.customize_exist_vm()
-            if cus_result['failed']:
-                return cus_result
+            # TODO not sure if it is possible to query the current customspec to compare against the one being provided to check in check mode.
```

Comment:
should TODO be addressed?

Reply:
There are already so many changes to the module that I think it best to keep this TODO as open. We can have a look at it if someone runs into a problem there.